### PR TITLE
Apply new structure, add additional content

### PR DESCRIPTION
**Pull request by JKRhb**
[Rendered Version](https://namib.me/draft-coap-mud/new-structure/draft-romann-mud-constrained.html)

### CODE REVIEW

File: draft-romann-mud-constrained.md [hunk 1]
```diff
@@ -72,7 +77,103 @@ Building upon the terminology defined in {{!RFC8520}}, this specification introd
...
+constrained environment can be seen in {{arch1-fig}}.
+Here, we can see that both the Thing and the MUD manager (as the recipient of
+the MUD URLs) may initiate the MUD discovery process:
+The Thing can contact and register with MUD URL recipients, e.g. by sending a
+CoAP POST request via Multicast, addressing a well-known registration endpoint.
+Conversely, MUD recipients can initiate the discovery process, e.g. by sending a
+COAP GET request to a well-known URI via multicast.
+Therefore, with the additional parameters introduced in this document, both
+Things and MUD managers can play a more active role during the MUD discovery
+process.
```
Comment (pulsastrix) [inline draft-romann-mud-constrained.md]: Repetition between lines 90-91 and 102-103

File: draft-romann-mud-constrained.md [hunk 2]
```diff
@@ -129,39 +247,27 @@ This recommendation will most likely be updated once a canonical encoding format
...
+<!-- ## Multicast -->
 
 {{!RFC7252}} registers one IPv4 and one IPv6 address each for the purpose of CoAP multicast.
 In addition to these already existing "All CoAP Nodes" multicast addresses, this document defines additional "All MUD CoAP Nodes" multicast addresses that can be used to address only the subset of CoAP Nodes that support MUD.
 If a device exposes a MUD URL via CoAP, it SHOULD join the respective multicast groups for the IP versions it supports.
 
 TODO: Add example
 
-# Obtaining a MUD URL in Constrained Environments
+## CoRE Link Format and CoRe Resource Direectories
```
Comment (pulsastrix) [inline draft-romann-mud-constrained.md]:
```suggestion
## CoRE Link Format and CoRE Resource Direectories
```

**Review by pulsastrix — CHANGES_REQUESTED** [review]

File: draft-romann-mud-constrained.md (hunk 1 above)
Comment (JKRhb) [reply draft-romann-mud-constrained.md]: I guess we can just remove the last sentence?

File: draft-romann-mud-constrained.md [hunk 3]
```diff
@@ -72,7 +77,103 @@ Building upon the terminology defined in {{!RFC8520}}, this specification introd
...
+Conversely, MUD recipients can initiate the discovery process, e.g. by sending a
+COAP GET request to a well-known URI via multicast.
+Therefore, with the additional parameters introduced in this document, both
+Things and MUD managers can play a more active role during the MUD discovery
+process.
+
+TODO: Refine diagram
+
+<!--
+TODO: Fix once plantuml is included in the GitHub Actions Docker image.
```
Comment (JKRhb) [inline draft-romann-mud-constrained.md:105]: This is probably not going to happen, by the way. I also tried out mermaid, but that didn't work either. I guess we will need to stick to some ASCII art then and/or generate figures separately. But this is probably not that relevant yet.

File: draft-romann-mud-constrained.md (hunk 1 above)
Comment (JKRhb) [reply draft-romann-mud-constrained.md]:
```suggestion
manager, both the Thing and the MUD manager can play more active roles when
using CoAP {{!RFC7252}} for exposing and discovering MUD URLs.

A general overview of the MUD architecture adjusted for using CoAP in a
constrained environment can be seen in {{arch1-fig}}.
Here, we can see that both the Thing and the MUD manager (as the recipient of
the MUD URLs) may initiate the MUD discovery process:
The Thing can contact and register with MUD URL recipients, e.g. by sending a
CoAP POST request via Multicast, addressing a well-known registration endpoint.
Conversely, MUD recipients can initiate the discovery process, e.g. by sending a
COAP GET request to a well-known URI via multicast.
```

**Pull request merged**